### ISSUE TITLE
Versioning info on API docs

### ISSUE THREAD

**Issue by apc518** [processing/p5.js#5746]
### Topic

I think it would be useful to have a minimum version note on each API doc (each method/constant/class page on p5js.org/reference)

The only problem with this that I can see is that it might not be worth the work to go document it all. Its pretty easy to just tell people to use a cdn script tag with the latest version, and if the versions had to be figured out manually that could be a lot of work.

But there's a lot I don't know, so I'm opening this discussion!

**Comment by limzykenneth** [comment]
Hi @apc518, I do see versioning info on some JS libraries and frameworks but in the case of p5.js I'm trying to think of a use case for it. 

For the most part the API of p5.js is very stable with not very complex code from early releases of p5.js will still likely work without modifications with the latest version, and even in the case that it doesn't I figure it will be faster to make some updates to the code instead of trying to track down the precise maximum version of p5.js that will work with it, exception being the case where the code for some reason couldn't be modified but the version of p5.js used could be but I feel that's a niche scenario.

Maybe for future releases everything can be tagged something like "available since 1.4.2" and we don't dig through the full history for a precise version number a particular method started from, but I also don't know of a particular use case for this.

If you have a particular use case for this feature, please do share! Thanks.

**Comment by apc518** [comment]
The specific issue I ran into that made me think about this was the function isLooping. I had an older version of p5 where that function didn't exist, so I was getting a reference error but was confused since other p5 functions were working fine and the docs said that isLooping was a function

**Comment by limzykenneth** [comment]
In this case, would having access to versioned API documentation be better? Since you would need to know your issue stem  from having an older version of p5.js anyway, you can continue to work against a whole documentation generated from the version you are using instead of mentally parsing version number each time you look up a function.

**Comment by apc518** [comment]
That sounds functionally the same, yea. Because if the api documentation was versioned like that, then I would see that I'm on the "version x.x.x" docs when I find a page for a function anyway

**Comment by limzykenneth** [comment]
Yes it would be more or less the same functionally but much much easier to implement. Though that being said it would probably still take some development time to do.

**Comment by Ayush4958** [comment]
Hi @ksen0 , @limzykenneth 

I read the issue thread and completely agree that doing this manually is huge maintenance burden, approach of hosting versioned api documentation make perfect sense is a scalable solution.

for implement this, I plan to dive into the repo to see how we can update the deployment pipeline to preserve historical reference snapshots (data.json) and will add ver switch to referrence page

I would like to work on this and get assigned for this issue, If any other contributor is not had taken over this

**Comment by ksen0** [comment]
@Ayush4958 thanks for the interest! IMO depending on exactly what the solution is for https://github.com/processing/p5.js-website/issues/432 this one could be a side-functionality of that (eg if offline docs artifact is generated based on a specific build of the website with a specific `data.json`, then it is relatively even easier to create downloadable docs for each version - addressing this need with relatively minimal additional maintenance). I think that would be better than a distinct additional pipeline, if possible. The offline docs issue is being worked on by a student group, and they will post their findings on the other issue this or next week. They are also on discord.p5js.org in #catalyst - if you want you can say hi there and say you'd like to work on this one? I am also in that channel, so as long as theres clarity on **if/how** these 2 issues are related you'd be welcome to work on it!

Context on why I'm attaching this to offline docs topic: I think a version switcher would be quite a bit of effort to maintain, and would need some more justification. I would prefer to avoid adding website features if possible. Therefore I am wondering if it's possible to make versioned docs available as archived artifacts; **if that is easier**. If it turns out that's also difficult and not meaningfully related to offline docs, we can revisit the switcher idea, but really thinking of it at the "what's the absolute most minimal useful way to make version-specific docs available to users?"

**Comment by Ayush4958** [comment]
Thanks for the detailed context, @ksen0!

I will hop over to the `#catalyst` channel. Once they share their findings on how they are generating the offline docs for the current version, we can coordinate on the easiest way to automate that generation for historical versions too.
